Move "running scans" output message earlier in the execution flow

nureka-rodrigo opened the pull request:
## Purpose

Fixes: https://github.com/ballerina-platform/static-code-analysis-tool/issues/75

In the original implementation, it took a considerable amount of time for the 'Running scans' text to appear in the console, because several time-consuming operations were performed beforehand.

1. Loading project configurations
2. Getting external analyzers
3. Loading platform plugins
4. Processing rules to include/exclude

## Approach

This change moved the "Running Scans" message much earlier in the execution flow, right after the basic validation but before the heavy operations.

<img width="774" height="557" alt="image" src="https://github.com/user-attachments/assets/19ef6610-3406-4f1d-b2c8-92190bb4076b" />

## Check List

- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Integration Tests

nureka-rodrigo committed:
Move "running scans" output message earlier in the execution flow

This change moved the "Running Scans" message much earlier in the execution flow, right after the basic validation but before the heavy operations.

nureka-rodrigo commented:
@azinneera @keizer619 PRAM

SasinduDilshara commented on scan-command/src/main/java/io/ballerina/scan/internal/ScanCmd.java:
```diff
@@ -185,6 +185,9 @@ public void execute() {
             return;
         }
 
+        outputStream.println();
+        outputStream.println("Running Scans");
```
```suggestion
        outputStream.println("\nRunning Scans");
```

Better we can add this as a constant variable and use

nureka-rodrigo replied:
Fix: 801dbf929204ea4f9a70c1d05fba7f03ba6d08e1

nureka-rodrigo replied:
@SasinduDilshara I think using `println()` instead of `\n` is the better approach. The `println()` method is platform-independent and explicitly terminates the line using the system’s line separator, which is safer than hardcoding `\n`. As the official Java [PrintStream](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/PrintStream.html) documentation states, 

>_Terminates the current line by writing the line separator string. The line separator string is defined by the system property `line.separator`, and is not necessarily a single newline character ('\n')._ 

While `\n` works on Unix-like systems, it may not represent the correct line ending on Windows or other environments. Using `println()` also makes the intent to end the line clearer to readers and avoids subtle inconsistencies in cross-platform code, which is why I think it is the better option here.

nureka-rodrigo committed:
Refactor output message for running scans

Updated the output message for running scans to use a constant from ScanToolConstants, improving maintainability and readability of the code.

gayaldassanayake commented on scan-command/src/main/java/io/ballerina/scan/internal/ScanToolConstants.java:
```diff
@@ -38,6 +38,7 @@ public class ScanToolConstants {
     static final String CODE_SMELL = "CODE_SMELL";
     static final String RULE_ID = "id";
     static final String RULE_DESCRIPTION = "description";
+    static final String RUNNING_SCANS = "Running Scans";
```
```suggestion
    static final String RUNNING_SCANS_LOG = "Running Scans";
```

nureka-rodrigo replied:
Fix: daa5516ab2d7fb5877ef452b99cb4239c408cdb2

nureka-rodrigo committed:
Rename constant RUNNING_SCANS to RUNNING_SCANS_LOG

Updated the constant name for improved clarity and consistency
